test/utils.jl

```diff
@@ -0,0 +1,136 @@
+using ADNLPModels, NLPModels, NLPModelsJuMP, OptimizationProblems
+
+function set_meta(all::AbstractVector{T}) where T <: Union{Symbol, String}
+  for name in string.(all)
+    io = open("src/Meta/" * name * ".jl", "w")
+    write(io, generate_meta(name))
+    close(io)
+  end
+end
+
+"""
+  `generate_meta(model, name)`
+  `generate_meta(name)`
+  
+  is used to generate the meta of a given NLPModel.
+
+Default values for undetermined entries:
+- `contype`: `:general` (`:linear` if `nlp.meta.nlin == nlp.meta.ncon`, or `unconstrained`)
+- `objtype`: `:other`
+- `cvx_obj`: `false`
+- `cvx_feasible_set`: `false` (`true` if unconstrained)
+- `cq`: `UInt8(0)`
+- `optimal_value`: `NaN`
+- `has_multiple_solutions`: `missing`
+- `everywhere_defined`: `missing`
+- `origin`: `:unknown`
+- `deriv`: `UInt8(0)`
+`is_infeasible` is `missing` unless the problem is unconstrained or `nlp.meta.x0` is feasible.
+"""
+function generate_meta(nlp::AbstractNLPModel, name::String)
+  contype = if nlp.meta.ncon == 0 && !(length(nlp.meta.ifree) < nlp.meta.nvar)
+    :unconstrained
+  elseif nlp.meta.nlin == nlp.meta.ncon > 0
+    :linear
+  else
+    :general
+  end
+  objtype = :other
+
+  cvx_obj = missing
+  cvx_feasible_set = nlp.meta.ncon == 0 ? true : missing
+
+  cq = UInt8(0)
+
+  origin = :unknown
+
+  default_nvar = ADNLPProblems.default_nvar
+  if name in ["clplatea", "clplateb", "clplatec", "fminsrf2"] # issue because variable is a matrix
+    variable_size, variable_con_size = false, false
+    nvar_formula, ncon_formula = "$(nlp.meta.nvar)", "$(nlp.meta.ncon)" # default values, to be modified
+  else
+    variable_size, variable_con_size, nvar_formula, ncon_formula = var_size(name, default_nvar)
+  end
+
+  str = "$(name)_meta = Dict(
+  :nvar => $(nlp.meta.nvar),
+  :variable_size => $(variable_size),
+  :ncon => $(nlp.meta.ncon),
+  :variable_con_size => $(variable_con_size),
+  :minimize => $(nlp.meta.minimize),
+  :name => \"$(name)\",
+  :has_equalities_only => $(length(nlp.meta.jfix) == nlp.meta.ncon > 0),
```

Comment:
Maybe we should have more meta information and avoid some of these `has` things. This would allow more customized filtering later.

Reply:
Things like the number of equalities, inequalities, etc.

Reply:
The problem with many of the meta-information is that they aren't flexible enough for scalable problems. 
That's the idea of the `get_name_meta(::Integer)` function.